build.yaml

```diff
@@ -715,6 +715,7 @@ filegroups:
   - src/cpp/common/channel_filter.h
   - src/cpp/server/dynamic_thread_pool.h
   - src/cpp/server/thread_pool_interface.h
```

Comment:
Why isn't thread_pool_interface.h (and related files) being deleted in this PR and from build.yaml, if the concept is indeed being deprecated?